### PR TITLE
Add backTo to many pages so it shows the correct screen below the RHP overlay

### PR DESCRIPTION
**Pull request by bernhardoj**
### Details

While we open a page from report RHP, refreshing will show the report screen below the report RHP instead of the search page. This PR fixes it.

### Fixed Issues

$ https://github.com/Expensify/App/issues/46773
PROPOSAL: https://github.com/Expensify/App/issues/46773#issuecomment-2267567434

### Tests

Same as QA Steps
- [x] Verify that no errors appear in the JS console

### Offline tests

Same as QA Steps
### QA Steps

Prerequisite: the account never dismisses hold educational page/modal yet
Web/Desktop
1. Submit a new expense
2. Open search page
3. Open the expense from step 1
4. Hold the expense. The hold educational page should show
5. Refresh the page
6. Verify the page behind the overlay is still the search page

Android/iOS/mWeb
Can't see page behind the RHP, so just make sure the hold educational modal still shows normally

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/dc167782-046c-4998-80fa-69bb706a6ad4

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/b20280f2-8e73-4229-8b23-14b04d163ca3

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/9d00843a-628e-4360-8ade-871cc232d952

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/d70f1152-c342-4b07-a388-1e8194cdf86d

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/ffa60091-f020-4fc5-b24a-f77f21c5f261

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/f9090ed3-7793-4f75-b371-1bbd55adacbd

</details>

### CODE REVIEW

**Comment by bernhardoj** [comment]
Additional video for other pages:

Report settings:

https://github.com/user-attachments/assets/3ce2ad2c-cbca-4956-945f-64b60bfe5c37

Money request view data:

https://github.com/user-attachments/assets/12babeea-3c99-45d6-ba89-354856c4ccfd

Profile from message:

https://github.com/user-attachments/assets/ef0ff83c-6be7-4bca-9110-3bd7af714cf8

Report field:

https://github.com/user-attachments/assets/688b6617-d102-4248-97f1-9c6bf6d94105

New task page:

https://github.com/user-attachments/assets/7adaa8b2-4c7e-4147-884f-c6714348a466

**Comment by bernhardoj** [comment]
For [this](https://github.com/Expensify/App/issues/46773#issuecomment-2273515567), I didn't add the `backTo` to the `goBack` fallback because it will goes back to the expense report and will open the hold educational page again on mount.
https://github.com/Expensify/App/blob/f58439cfb116abd338dae880088a26725794108e/src/pages/ProcessMoneyRequestHoldPage.tsx#L38

1. Open expense report/thread
2. Hold expense
3. Hold educational shows (stack: [expense report, hold educational])
4. Refresh (stack: [hold educational])
5. Go back. expense report/thread is mounted (stack: [expense report])
6. This logic is triggered
https://github.com/Expensify/App/blob/f58439cfb116abd338dae880088a26725794108e/src/components/MoneyRequestHeader.tsx#L99-L118

which opens the hold educational again.

File: src/components/MoneyRequestConfirmationListFooter.tsx
Comment (luacmartins) [inline src/components/MoneyRequestConfirmationListFooter.tsx:285]:
```suggestion
                        Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_AMOUNT.getRoute(action, iouType, transactionID, reportID, CONST.MODAL.MODAL_TYPE.CONFIRM, Navigation.getActiveRouteWithoutParams()));
```

**Review by luacmartins — COMMENTED** [review]
Left a small comment

File: src/components/MoneyRequestConfirmationListFooter.tsx
Comment (bernhardoj) [reply src/components/MoneyRequestConfirmationListFooter.tsx]: Done

**Comment by bernhardoj** [comment]
After merging with main, the page without params will now return undefined if we access `route.params`. It was an empty object before. So, I updated some pages to use null-safety.

**Review by luacmartins — DISMISSED** [review]
LGTM

**Comment by bernhardoj** [comment]
> Note that in https://github.com/Expensify/App/pull/47690 we will be adding the chat type to search, therefore more report-related routes will be accessible from RHP, therefore I think we need to add the backTo param to more routes like:

I would prefer to wait for it to be merged so I can test it.

In the meantime, I'll add more backTo to these pages
> REPORT_AVATAR
ATTACHMENTS
REPORT_WITH_ID_DETAILS_EXPORT
TRANSACTION_RECEIPT
TRANSACTION_DUPLICATE_* (All)
and for create flows (submit expense, track expense...) like you've done already with "assign task"

**Comment by bernhardoj** [comment]
1. Adding `backTo` for these pages doesn't help. `backTo` is currently only for RHP.
REPORT_AVATAR
ATTACHMENTS
TRANSACTION_RECEIPT

2. **Done**
REPORT_WITH_ID_DETAILS_EXPORT
TRANSACTION_DUPLICATE_* (All)

> and for create flows (submit expense, track expense...) like you've done already with "assign task"

3. First of all, only submit and track is available on expense report. Adding `backTo` for these pages is currently "useless". As you can see on the video below, when the page is mounted, the tab screen overrides the route (internally done by the tab navigator) which doesn't contains the `backTo`. So, refreshing the page will show report screen instead of search page.

https://github.com/user-attachments/assets/172d259c-e130-475b-b254-db950cdb527f

**Review by rayane-d — CHANGES_REQUESTED** [review]
Let's add `backTo` param to `FLAG_COMMENT` route

**Comment by luacmartins** [comment]
@bernhardoj we also have conflicts

**Comment by bernhardoj** [comment]
Done

**Comment by luacmartins** [comment]
We're close to getting https://github.com/Expensify/App/pull/47690 merged, so we can update this PR

**Comment by luacmartins** [comment]
@bernhardoj we have conflicts. The holding PR is also merged

**Comment by bernhardoj** [comment]
I'm still adding the rest of the backTo to the pages that are accessible from chats. I'll push them all at once.

**Comment by bernhardoj** [comment]
@rayane-djouah pushed new commit. Please check and let me know if I missed anything!

**Comment by luacmartins** [comment]
@bernhardoj we have conflicts

**Comment by luacmartins** [comment]
@rayane-djouah conflicts are resolved. Let's resume review.

**Review by rayane-d — CHANGES_REQUESTED** [review]

**Comment by luacmartins** [comment]
@bernhardoj we have conflicts again

**Comment by bernhardoj** [comment]
Conflict solved

**Comment by luacmartins** [comment]
Thanks! Let's get this to the finish line @rayane-djouah

**Comment by bernhardoj** [comment]
Fixed both.

**Review by rayane-d — CHANGES_REQUESTED** [review]

File: src/components/MoneyReportHeader.tsx
Comment (bernhardoj) [reply src/components/MoneyReportHeader.tsx]: Oops, you're right. Thanks for catching that! Reverted.

**Review by rayane-d — APPROVED** [review]
LGTM and tests well 👍

**Review by rayane-d — CHANGES_REQUESTED** [review]
Oops! Just found a bug:

https://github.com/user-attachments/assets/94090364-c7c4-4eb3-a9a6-fdb63b677152

**Comment by bernhardoj** [comment]
I think we can keep it optional, otherwise I need to re-check all ROUTES to see which backTo can be optional or not. backTo by pattern is also optional.

> Oops! Just found a bug:

Fixed.

**Review by rayane-d — APPROVED** [review]
LGTM and tests well 👍

**Comment by bernhardoj** [comment]
Fixed

**Comment by luacmartins** [comment]
Those tests seem broken on main too. Seeing the same on other PRs.

**Review by rayane-d — APPROVED** [review]

**Comment by bernhardoj** [comment]
@luacmartins conflicts fixed

**Review by luacmartins — APPROVED** [review]

**Comment by luacmartins** [comment]
Yes, we can keep the backTo param optional. ESLint errors are unrelated. Merging.

**Pull request merged**
